### ISSUE TITLE
This module ignored the "ansible_host" option in ansible inventory files.

### ISSUE THREAD

**Issue by RuiMeireles** [Juniper/ansible-junos-stdlib#378]
Issue Type
------

 - Bug Report

Module Name
------

All modules in ansible-junos-stdlib have this issue

Juniper.Junos role and Python libraries version

```
(venv) neteng-ansible-network [rmeireles] $ ansible --version
ansible 2.5.0
  config file = /Users/rmeireles/git/neteng-ansible-network/ansible.cfg
  configured module search path = [u'/Users/rmeireles/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/rmeireles/git/neteng-ansible-network/venv/lib/python2.7/site-packages/ansible
  executable location = /Users/rmeireles/git/neteng-ansible-network/venv/bin/ansible
  python version = 2.7.14 (default, Jan  6 2018, 12:16:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]

(venv) neteng-ansible-network [rmeireles] $ pip freeze
ansible==2.5.0
asn1crypto==0.24.0
bcrypt==3.1.4
boto==2.47.0
boto3==1.4.1
botocore==1.4.62
cffi==1.11.5
cryptography==2.2.2
docutils==0.14
enum34==1.1.6
futures==3.2.0
idna==2.7
ipaddress==1.0.22
Jinja2==2.10
jmespath==0.9.3
junos-eznc==2.1.8
jxmlease==1.0.1
lxml==4.2.1
MarkupSafe==1.0
ncclient==0.5.3
netaddr==0.7.19
paramiko==2.4.1
pyasn1==0.4.3
pycparser==2.18
PyNaCl==1.2.1
pyserial==3.4
python-dateutil==2.7.3
PyYAML==3.12
s3transfer==0.1.13
scp==0.11.0
six==1.10.0
urllib3==1.22


Using Juniper.junos 2.1.0
```

OS / Environment
------

Junos version: 12.3X48-D35.7
Model: SRX550

Summary
------

I experienced a problem running all modules in ansible-junos. 
I kept getting this error, even though the credentials were OK, and PyEZ was working when I tested it via a Python shell:
```
 "msg": "Unable to make a PyEZ connection: ConnectUnknownHostError(lab-fw01-sc1)"
```
I figured out that the problem is that I had this in my inventory file. I was referring my host by a short name (lab-fw01-sc1), and your module seems to be trying to establish connections to that short name, instead of using the full name in ansible_host.
```
[junos_firewalls]
lab-fw01-sc1 ansible_host=lab-fw01-sc1.net.atlassian.com
```

You should update the files so that this is handled properly.

Steps to reproduce
------

Create a inventory entry where the host name is not the FQDM of the device:
```
[junos_firewalls]
lab-fw01-sc1 ansible_host=lab-fw01-sc1.net.atlassian.com
```

```yaml

```

Expected results

```

```
Actual results
------

```

```

**Comment by dgsardina** [comment]
Same issue, in my case I do not have any DNS registry to the junos nodes neither in the /etc/hosts so I use `[junos]
fw ansible_host=10.10.10.1
`

**Comment by aristima** [comment]
I cannot comment the potential issue, but we are also using ansible_host option for example in cases where we need to access the equipment via management Ethernet instead of in-band SSH or vice versa. We have worked around this by specifying in the provider:

provider:
  host: "{{ ansible_host | default(inventory_hostname) }}"

This works perfectly at least for our needs.

**Comment by cjguerrero** [comment]
I am running into this same issue with an EX9208. Where did you specify the provider argument (i.e. in a vars file, playbook, etc)?

**Comment by aristima** [comment]
I have the provider variable defined in group_vars. It used to be in the playbook but I then moved it to group_vars so that it is automatically used for all Juniper equipment, in our case 'all-routers'

**Comment by Alex-Ignatov** [comment]
We are facing the same issue when running playbooks against EX switches... This issue has been open for a while - any chance it will be resolved anytime soon?

**Comment by tc0nn** [comment]
Same issue. Adding my hostname from my inventory to the /etc/hosts files seems to fix it also.

**Comment by rsmekala** [comment]
The root cause of the is found.

In the below line, if the preference order is incorrect:
https://github.com/Juniper/ansible-junos-stdlib/blob/258bbb0d4df02e1458c0ecf4dd69f67550e6dc2f/module_utils/juniper_junos_common.py#L513
It should be:
```
'host': ['ansible_host', 'inventory_hostname'],
```

In case, both variables are provided we should prefer host value over hostname. Will raise a pull request for the same.

Please expect a delay in merging the PR, as Travis checks are currently failing.

**Comment by rsmekala** [comment]
Fixed this via #453. Please verify and reopen if necessary.
